editoast/src/models/stdcm_search_environment.rs

```diff
@@ -5,14 +5,49 @@ use diesel::ExpressionMethods;
 use diesel::QueryDsl;
 use diesel_async::RunQueryDsl;
 use editoast_derive::Model;
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
 use std::ops::DerefMut;
 use utoipa::ToSchema;
 
 use editoast_models::prelude::*;
 
-#[cfg(test)]
-use serde::Deserialize;
+#[editoast_derive::openapi_schema]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default, ToSchema)]
+pub struct OptionalArrayOfInt(Option<Vec<i64>>);
+
+impl OptionalArrayOfInt {
+    pub fn new(value: Option<Vec<i64>>) -> Self {
+        Self(value)
+    }
+
+    pub fn to_vec(&self) -> Option<Vec<i64>> {
+        self.0.as_ref().map(|vec| vec.clone().into_iter().collect())
```

Comment:
Isn't the following not working?

```suggestion
        self.0.clone()
```

It doesn't make sense to transform a `Vec` into an iterator, then collect it back into a `Vec`.
It might need to be fixed in `Tags` too (since I see from where you took inspiration 😄)

Reply:
Done